library/core/src/iter/adapters/zip.rs

```diff
@@ -295,9 +295,10 @@ where
                 let sz_a = self.a.size();
                 if A::MAY_HAVE_SIDE_EFFECT && sz_a > self.len {
                     for _ in 0..sz_a - self.len {
+                        self.a_len -= 1;
```

Comment:
Might be a good idea to have a comment explaining why we do this in steps of 1.

Reply:
This kind of pattern occurs in many places, e.g. the family of methods that fill/append to vecs from iterators and updates lengths. It's a common pitfall to just update the length after iteration is done, it seems more efficient after all when you don't think about the panics and that you're running user-supplied code.

So I'm not sure if adding a comment to this one place among many will really improve anything. But if you still think it's valuable I can do it anyway.

Reply:
How many is many?

> It's a common pitfall to just update the length after iteration is done

Exactly. That's why I think a comment is in order. Don't assume the next person reading this code knows as much as you do, or has read all the other files in this folder.

Reply:
There's about 4 of those in vec, in various flavors. Some use drop guards and bump pointers instead of setting length directly. Of the reading type there's 1-4 cases in `Zip`, depending on whether one wants to count only loops or single step ones. One of the single-step ones has been wrong before (#81740). Another one in `core::array`, but that one already has plenty of comments. Possibly with more to come in the future as more const-generic methods are added to fill arrays. Haven't checked other types.

Reply:
Still seems like an amount where adding (brief) comments individually doesn't seem overwhelming.

Reply:
Ok, will do.

Reply:
Thanks. :)